### PR TITLE
fix: blockscale moe routine supports non-DS routing

### PR DESCRIPTION
**Pull request by jdebache**
## 📌 Description

The comment at this location previously stated that only `float32` routing logits would be possible for blockscale MoE. However, it seems the launcher now supports bfloat16 for non-DS routing.

## Bug Fixes
* Per-method routing precision handling updated: the model now selects routing tensor precision based on the chosen routing method (e.g., DeepSeekV3 uses higher-precision), replacing fixed per-path precision. This improves routing stability and consistency across routing configurations without changing public APIs.

### CODE REVIEW

File: benchmarks/routines/moe.py [hunk 1]
```diff
@@ -281,10 +281,14 @@ def create_trtllm_moe_test_data(
...
-            (num_tokens, num_experts), device=device, dtype=torch.float32
-        )
+        if routing_method_type == 2:  # DeepSeekV3 - uses float32
+            routing_logits = torch.randn(
+                (num_tokens, num_experts), device=device, dtype=torch.float32
+            )
+        else:
+            routing_logits = torch.randn(
+                (num_tokens, num_experts), device=device, dtype=torch.bfloat16
+            )
```
[the remark this answers is not in the document]
Comment (bkryu) [reply benchmarks/routines/moe.py]: Gemini's suggestion here indeed looks cleaner. Do you mind reflecting this?

[the remark this answers is not in the document]
Comment (jdebache) [reply benchmarks/routines/moe.py]: I applied it to the three branches.

**Review by bkryu — APPROVED** [review]
LGTM
